src/main/scala/com/ing/wbaa/airlock/proxy/provider/MessageProviderKafka.scala

```diff
@@ -0,0 +1,34 @@
+package com.ing.wbaa.airlock.proxy.provider
+
+import akka.Done
+import com.ing.wbaa.airlock.proxy.data._
+import com.ing.wbaa.airlock.proxy.provider.kafka.EventProducer
+import akka.http.scaladsl.model.{ HttpMethod, HttpMethods, RemoteAddress }
+import com.ing.wbaa.airlock.proxy.provider.aws.{ s3ObjectCreated, s3ObjectRemoved }
+
+import scala.concurrent.Future
+
+trait MessageProviderKafka extends EventProducer with AWSMessageEventJsonSupport {
+
+  private def incompeteData: Future[Nothing] = Future.failed(throw new Exception("Cannot send event to kafka, not enough input data"))
```

Comment:
The method `Future.failed` accepts an instance of `Throwable`, hence the exception should not be thrown but simply passed to the method (i.e. remove the keyword `throw`).